src/index.ts

```diff
@@ -138,19 +138,28 @@ export const createInstance = (config: Config): Instance => {
 
       return tx;
     },
+
     cancelBid: async (
       auctionId: string,
-      signer: ethers.Signer
-    ): Promise<ethers.ContractTransaction> => {
-      const zAuction = await getZAuctionContract(
-        signer,
-        config.zAuctionAddress
-      );
-
-      const account = await signer.getAddress();
-
-      const tx = await zAuction.cancelBid(account, auctionId);
-      return tx;
+      signedBidMessage: string,
+      cancelOnChain: boolean,
+      signer?: ethers.Signer
+    ) => {
+      // Always cancel the bid through the API
+      const encodedCancelMessage = await apiClient.encodeCancelBid(signedBidMessage);
+      await apiClient.submitCancelBid(encodedCancelMessage, signedBidMessage)
```

Comment:
API returns a 400 but I've added a case for this in the SDK code that passes a `softFail` boolean to the `makeApiCall` function so that if no bid is found it doesn't fail and lets the contract cancel the bid.